File: 2015/day05/second.py
f = open("input.txt")
lines = f.readlines()

sum = 0

for line in lines:
	g1 = False
	g2 = False
	for i in range (0, len(line)-2):
		if line[i] == line[i+2]:
			g1 = True

		if len(line) > 3 and (line.count(line[i:i+2]) > 1):
			g2 = True
	
	if g1 and g1 == g2:
		sum+=1			

print(sum)

File: 2022/03/second.go
package main

import (
	"bufio"
	"log"
	"os"
	"strings"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		log.Fatalf("can not open file")
	}

	defer file.Close()

	items := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

	scanner := bufio.NewScanner(file)
	sum := 0
	rs := []string{}
	index := 1

	for scanner.Scan() {
		rs = append(rs, scanner.Text())
		if index > 0 && index%3 == 0 {
			//log.Println(rs)
			for _, c := range items {
				if strings.Contains(rs[0], string(c)) && strings.Contains(rs[1], string(c)) &&
					strings.Contains(rs[2], string(c)) {
					sum += strings.Index(items, string(c)) + 1
				}
			}
			index = 0
			rs = []string{}
		}
		index++
	}

	log.Println(sum)

}
